src/test/java/stepdefs/SearchStepDefs.java

```diff
@@ -35,19 +36,47 @@ public void iAmOnTheGoogleUkHomepage() {
     driver.manage().window().maximize();
     driver.get("https://www.google.co.uk");
   }
-  
-  @When("I enter a search term")
-  public void iEnterASearchTerm() {
+
+  @Given("^I am on the Google UK$")
+  public void iAmOnTheGoogleUk() {
+    driver.manage().window().maximize();
+    driver.get("https://www.google.co.uk");
+  }
+  @When("I enter {string} search term")
+  public void iEnterSearchTerm(String searchItem) {
+
     WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
-    wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@id='L2AGLb2']"))).click();
-    driver.findElement(By.xpath("//input[@title='Search']")).sendKeys("BBC news");
-    driver.findElement(By.xpath("//input[@title='Search']")).sendKeys(Keys.ENTER);
+    GoogleHomePage.CookiesButton_search(driver).click();
+    GoogleHomePage.textBox_search(driver).sendKeys(searchItem);
+    //wait.until(ExpectedConditions.elementToBeClickable(By.name("btnK"))).sendKeys(Keys.ENTER);
+    wait.until(ExpectedConditions.elementToBeClickable(GoogleHomePage.GoogleButton_search(driver))).sendKeys(Keys.ENTER);
+    GoogleHomePage.GoogleButton_search(driver).click();
+
+  }
+
+  @When("I enter {string} in lucky search term")
+  public void iEnterInLuckySearchTerm(String searchItem) {
+
+    WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));
+    GoogleHomePage.CookiesButton_search(driver).click();
+    GoogleHomePage.textBox_search(driver).sendKeys(searchItem);
+    //wait.until(ExpectedConditions.elementToBeClickable(By.name("btnI"))).sendKeys(Keys.ENTER);//
+    wait.until(ExpectedConditions.elementToBeClickable(GoogleHomePage.LuckyButton_search(driver))).sendKeys(Keys.ENTER);
+    GoogleHomePage.LuckyButton_search(driver).click();
   }
 
   @Then("results relevant to the search term are returned")
   public void resultsRelevantToTheSearchTermAreReturned() {
+   List<WebElement> resultHeaders = driver.findElements(By.xpath("//a/h3"));
+   for (WebElement header : resultHeaders) {
+     assertThat(header.getText()).as("Search results contains search term").contains("BBC");
+    }}
+
+
+  @Then("results to the lucky click are returned")
+  public void resultsToTheLuckyClickAreReturned() {
     List<WebElement> resultHeaders = driver.findElements(By.xpath("//a/h3"));
-    for(WebElement header : resultHeaders) {
+    for (WebElement header : resultHeaders) {
```

Comment:
Instead of line 78 and 79, I would simply assert that the current Url contains 'BBC', so instead of asserting the header text contains bbc on line 80, just replace this with a 1 line assertion that gets the current url and checks it has BBC in it

Reply:
@Then("results relevant to the search term are returned")
  public void resultsRelevantToTheSearchTermAreReturned() {

    // Then the browser should redirect to a URL containing "BBC"
    assertThat(driver.getCurrentUrl()).as("...").contains("BBC");
  }

  @Then("results to the lucky click are returned")
  public void resultsToTheLuckyClickAreReturned() {

    // Then the browser should redirect to a URL containing "BBC"
    assertThat(driver.getCurrentUrl()).as("...").contains("bbc");

    }
  }